api/src/main/java/marquez/db/OpenLineageDao.java

```diff
@@ -104,37 +113,155 @@ default UpdateLineageRow updateBaseMarquezModel(LineageEvent event, ObjectMapper
             DEFAULT_NAMESPACE_OWNER);
     bag.setNamespace(namespace);
 
-    String description = null;
-    if (event.getJob().getFacets() != null
-        && event.getJob().getFacets().getDocumentation() != null) {
-      description = event.getJob().getFacets().getDocumentation().getDescription();
-    }
+    String description =
+        Optional.ofNullable(event.getJob().getFacets())
+            .map(JobFacet::getDocumentation)
+            .map(DocumentationJobFacet::getDescription)
+            .orElse(null);
 
     Map<String, String> context = buildJobContext(event);
     JobContextRow jobContext =
         jobContextDao.upsert(
             UUID.randomUUID(), now, Utils.toJson(context), Utils.checksumFor(context));
     bag.setJobContext(jobContext);
 
-    String location = null;
-    if (event.getJob().getFacets() != null
-        && event.getJob().getFacets().getSourceCodeLocation() != null) {
-      location = getUrlOrNull(event.getJob().getFacets().getSourceCodeLocation().getUrl());
-    }
+    String location =
+        Optional.ofNullable(event.getJob().getFacets())
+            .flatMap(f -> Optional.ofNullable(f.getSourceCodeLocation()))
+            .flatMap(s -> Optional.ofNullable(s.getUrl()))
+            .orElse(null);
 
+    Instant nominalStartTime =
+        Optional.ofNullable(event.getRun().getFacets())
+            .flatMap(f -> Optional.ofNullable(f.getNominalTime()))
+            .map(NominalTimeRunFacet::getNominalStartTime)
+            .map(t -> t.withZoneSameInstant(ZoneId.of("UTC")).toInstant())
+            .orElse(null);
+    Instant nominalEndTime =
+        Optional.ofNullable(event.getRun().getFacets())
+            .flatMap(f -> Optional.ofNullable(f.getNominalTime()))
+            .map(NominalTimeRunFacet::getNominalEndTime)
+            .map(t -> t.withZoneSameInstant(ZoneId.of("UTC")).toInstant())
+            .orElse(null);
+
+    Logger log = LoggerFactory.getLogger(OpenLineageDao.class);
```

Comment:
I would annotate the `OpenLineageDao` class with `@Slf4j`

Reply:
Can't - its an interface and `@Slf4j` only works on classes

Reply:
Ah you're right